scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,386 @@
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs bufr (cloud, metar, lightning) preprocess
+with FV3 for the specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=$(date +"%y%j%H" -d "${START_DATE}")
+PREYYJJJHH=$(date +"%y%j%H" -d "${START_DATE} 1 hours ago")
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for BUFR obseration process ..."
+
+cd_vrfy ${DATA}
+
+fixgriddir=$FIXgsi/${PREDEF_GRID_NAME}
+
+print_info_msg "$VERBOSE" "fixgriddir is $fixgriddir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixgriddir}/fv3_grid_spec          fv3sar_grid_spec.nc
+
+#-----------------------------------------------------------------------
+#
+#   copy bufr table
+#
+#-----------------------------------------------------------------------
+BUFR_TABLE=${FIXgsi}/prepobs_prep_RAP.bufrtable
+cp_vrfy $BUFR_TABLE prepobs_prep.bufrtable
+
+#-----------------------------------------------------------------------
+#
+#   set observation soruce 
+#
+#-----------------------------------------------------------------------
+if [[ "${NET}" = "RTMA"* ]]; then
+  SUBH=$(date +%M -d "${START_DATE}")
+  obs_source="rtma_ru"
+  obsfileprefix=${obs_source}
+  obspath_tmp=${OBSPATH}/${obs_source}.${YYYYMMDD}
+
+else
+  SUBH=""
+  obs_source=rap
+  if [[ ${HH} -eq '00' || ${HH} -eq '12' ]]; then
+    obs_source=rap_e
+  fi
+
+  case $MACHINE in
+
+  "WCOSS2")
+
+    obsfileprefix=${obs_source}
+    obspath_tmp=${OBSPATH}/${obs_source}.${YYYYMMDD}
+
+    ;;
+  *)
+
+    obsfileprefix=${YYYYMMDDHH}.${obs_source}
+    obspath_tmp=${OBSPATH}
+
+  esac
+fi
+
+#
+#-----------------------------------------------------------------------
+#-----------------------------------------------------------------------
+#
+# Link to the observation lightning bufr files
+#
+#-----------------------------------------------------------------------
+
+run_lightning=false
```

Comment:
For lightning bufr files, `run_lightning=false` is set first, while for NASA LaRC and METAR cloud bufr files, `run_cloud` and `run_metar` are set just before `if [ -r "${obs_file}" ]; then`.  While I'm okay with this, would it be best to move `run_lightning=false` to line 146 to make all three entries more similar?